yt_dlp/extractor/voicy.py

```diff
@@ -0,0 +1,150 @@
+            'title': '1/21(木)声日記：ついに原稿終わった！！',
+            'uploader': 'ちょまど@ ITエンジニアなオタク',
+            'uploader_id': '7339',
+        },
+        'playlist_mincount': 9,
+    }]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        assert mobj
+        voice_id = mobj.group('id')
+        channel_id = mobj.group('channel_id')
+        url, article_list = unsmuggle_url(url)
+        if not article_list:
+            article_list = self._call_api(self.ARTICLE_LIST_API_URL % (channel_id, voice_id), voice_id)
+        return self._extract_from_playlist_data(article_list)
+
+
+class VoicyChannelIE(VoicyBaseIE):
+    IE_NAME = 'voicy:channel'
+    _VALID_URL = r'https?://voicy\.jp/channel/(?P<id>\d+)'
+    PROGRAM_LIST_API_URL = 'https://vmw.api.voicy.jp/program_list/all?channel_id=%s&limit=20&public_type=3%s'
+    _TESTS = [{
+        'url': 'https://voicy.jp/channel/1253/',
+        'info_dict': {
+            'id': '7339',
+            'title': 'ゆるふわ日常ラジオ #ちょまラジ',
+            'uploader': 'ちょまど@ ITエンジニアなオタク',
+            'uploader_id': '7339',
+        },
+        'playlist_mincount': 54,
+    }]
+
+    @classmethod
+    def suitable(cls, url):
+        return not VoicyIE.suitable(url) and super(VoicyChannelIE, cls).suitable(url)
+
+    def _entries(self, channel_id):
+        pager = ''
+        for count in itertools.count(1):
+            article_list = self._call_api(self.PROGRAM_LIST_API_URL % (channel_id, pager), channel_id, note='Paging #%d' % count)
+            playlist_data = article_list.get('PlaylistData')
+            if not playlist_data:
+                break
+            yield from playlist_data
+            last = playlist_data[-1]
+            pager = '&pid=%d&p_date=%s&play_count=%s' % (last['PlaylistId'], last['Published'], last['PlayCount'])
+
+    def _real_extract(self, url):
+        channel_id = self._match_id(url)
+        articles = self._entries(channel_id)
+
+        first_article = next(articles, None)  # need the first item to get some info
+        title = traverse_obj(first_article, ('ChannelName', ), expected_type=compat_str)
+        if not title:
+            spaker_name = traverse_obj(first_article, ('SpeakerName', ), expected_type=compat_str)
+            if spaker_name:
+                title = 'Uploads from %s' % spaker_name
+        if not title:
+            title = 'Uploads from channel ID %s' % channel_id
+
+        articles = itertools.chain([first_article], articles) if first_article else articles  # push back first item
+
+        playlist = (  # intentionally use () to keep lazyness
+            self.url_result(smuggle_url('https://voicy.jp/channel/%s/%d' % (channel_id, value['PlaylistId']), value), VoicyIE.ie_key())
+            for value in articles)
+        return {
+            '_type': 'playlist',
+            'entries': playlist,
+            'id': channel_id,
+            'title': title,
+            'channel': channel_id,
```

Comment:
Shouldn't this be either the speaker_name or channel name?
Maybe channel=channel name, uploader=speaker_name

Reply:
For Voicy, channel name and uploader name (in yt-dlp's context) are considered same